storagenode/gracefulexit/worker.go

```diff
@@ -5,47 +5,226 @@ package gracefulexit
 
 import (
 	"context"
+	"io"
+	"os"
 	"time"
 
 	"github.com/zeebo/errs"
 	"go.uber.org/zap"
 
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
 	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/pieces"
+	"storj.io/storj/storagenode/piecestore"
 	"storj.io/storj/storagenode/satellites"
+	"storj.io/storj/storagenode/trust"
+	"storj.io/storj/uplink/ecclient"
 )
 
 // Worker is responsible for completing the graceful exit for a given satellite.
 type Worker struct {
 	log         *zap.Logger
-	satelliteID storj.NodeID
+	store       *pieces.Store
 	satelliteDB satellites.DB
+	trust       *trust.Pool
+	dialer      rpc.Dialer
+	satelliteID storj.NodeID
+	ecclient    ecclient.Client
 }
 
 // NewWorker instantiates Worker.
-func NewWorker(log *zap.Logger, satelliteDB satellites.DB, satelliteID storj.NodeID) *Worker {
+func NewWorker(log *zap.Logger, store *pieces.Store, satelliteDB satellites.DB, trust *trust.Pool, dialer rpc.Dialer, satelliteID storj.NodeID) *Worker {
 	return &Worker{
 		log:         log,
-		satelliteID: satelliteID,
+		store:       store,
 		satelliteDB: satelliteDB,
+		trust:       trust,
+		dialer:      dialer,
+		satelliteID: satelliteID,
+		ecclient:    ecclient.NewClient(log, dialer, 0),
 	}
 }
 
 // Run calls the satellite endpoint, transfers pieces, validates, and responds with success or failure.
 // It also marks the satellite finished once all the pieces have been transferred
-func (worker *Worker) Run(ctx context.Context, satelliteID storj.NodeID, done func()) (err error) {
+// TODO handle transfers in parallel
+func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	defer done()
+
 	worker.log.Debug("running worker")
 
-	// TODO actually process the order limits
-	// https://storjlabs.atlassian.net/browse/V3-2613
+	addr, err := worker.trust.GetAddress(ctx, worker.satelliteID)
+	if err != nil {
+		return errs.Wrap(err)
```

Comment:
nit - wondering if we should have a worker specific error class and use that to wrap instead of the general errs.Wrap.